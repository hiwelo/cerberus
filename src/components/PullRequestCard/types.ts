import { PullRequest } from 'types/GitHubApi/PullRequest';

export interface PullRequestCardProps {
  item: PullRequest;
}
