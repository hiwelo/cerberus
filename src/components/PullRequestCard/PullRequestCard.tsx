import React, { FunctionComponent } from 'react';
import { Link } from '..';
import { PullRequestCardProps } from './types';
import {
  Card,
  CardContent,
  CardHeader,
  CardHeaderSuffix,
  CardInfoLabel,
  CardInfoList,
  CardInfoValue,
} from './components';

export const PullRequestCard: FunctionComponent<PullRequestCardProps> = ({
  item,
}) => {
  return (
    <>
      <Card>
        <CardHeader>
          <Link href={item.url} target="_blank">
            {item.title}
            <CardHeaderSuffix>#{item.number}</CardHeaderSuffix>
          </Link>
        </CardHeader>
        <CardContent>{item.bodyText.slice(0, 140)}â€¦</CardContent>
        <CardInfoList>
          <CardInfoLabel>Author</CardInfoLabel>
          <CardInfoValue>
            <Link href={item.author.url} target="_blank">
              {item.author.login}
            </Link>
          </CardInfoValue>
          <CardInfoLabel>Deployments</CardInfoLabel>
          {item.commits?.edges.map(({ node }) => {
            const { commit } = node;

            return (
              <>
                {commit.deployments?.edges.map(({ node }) => (
                  <>
                    <CardInfoValue>
                      <Link
                        href={node.latestStatus.environmentUrl}
                        target="_blank"
                      >
                        {node.description}
                      </Link>
                    </CardInfoValue>
                  </>
                ))}
              </>
            );
          })}
        </CardInfoList>
      </Card>
    </>
  );
};
