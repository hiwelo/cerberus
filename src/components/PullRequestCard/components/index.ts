export { default as Card } from './Card';
export { default as CardHeader } from './CardHeader';
export { default as CardHeaderSuffix } from './CardHeaderSuffix';
export { default as CardContent } from './CardContent';
export { default as CardInfoLabel } from './CardInfoLabel';
export { default as CardInfoList } from './CardInfoList';
export { default as CardInfoValue } from './CardInfoValue';
export { default as CardLink } from './CardLink';
export { default as CardTitle } from './CardTitle';
