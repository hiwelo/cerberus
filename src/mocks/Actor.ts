import { Actor } from 'types/GitHubApi/Actor';

/**
 * Mocked Actor object
 */
export const mockedActor: Actor = {
  avatarUrl: 'https://i.pravatar.cc/300',
  login: 'test',
  url: 'https://github.com/test',
};
