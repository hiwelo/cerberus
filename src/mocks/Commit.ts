import { Commit } from '../types/GitHubApi/Commit';

/**
 * Mocked Commit object
 */
export const mockedCommit: Commit = {
  abbreviatedOid: '7638417',
  message: 'Test commit',
  messageBody: 'Test of message for the commit body',
  messageHeadline: 'Test of headline for the commit message',
  oid: '7638417db6d59f3c431d3e1f261cc637155684cd',
};
