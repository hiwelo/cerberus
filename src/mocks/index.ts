export { mockedActor } from './Actor';
export { mockedCommit } from './Commit';
export { mockedDeployment } from './Deployment';
export { mockedPullRequest } from './PullRequest';
