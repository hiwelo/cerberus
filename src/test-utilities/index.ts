export { renderWithApp } from './renderWithApp';
export { testSnapshot } from './testSnapshot';
