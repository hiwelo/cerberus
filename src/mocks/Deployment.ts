import { Deployment, DeploymentConnection } from 'types/GitHubApi/Deployment';
import { mockedActor } from './Actor';
import { mockedCommit } from './Commit';

export const mockedDeployment: Deployment = {
  commit: mockedCommit,
  commitOid: '7638417db6d59f3c431d3e1f261cc637155684cd',
  creator: mockedActor,
  description: 'Test deployment',
  environment: 'test',
  latestStatus: {
    creator: mockedActor,
    description: 'Test environment',
    environmentUrl: 'https://example.com/',
    state: 'SUCCESS',
    updatedAt: '2019-12-20T20:11:45+00:00',
  },
  state: 'ACTIVE',
  task: 'deploy',
  updatedAt: '2019-12-20T20:11:45+00:00',
};

export const mockedDeploymentConnection: DeploymentConnection = {
  edges: [
    {
      __typename: 'DeploymentEdge',
      node: mockedDeployment,
    },
  ],
};
