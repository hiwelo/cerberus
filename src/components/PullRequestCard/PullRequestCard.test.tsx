import React, { FunctionComponent } from 'react';
import { testA11y, testSnapshot } from '../../test-utilities';
import { mockedPullRequest } from '../../mocks';
import { PullRequestCard } from './PullRequestCard';

const TestComponent: FunctionComponent = () => (
  <PullRequestCard item={mockedPullRequest} />
);

describe('PullRequestCard', () => {
  testSnapshot(<TestComponent />);
  testA11y(<TestComponent />);
});
