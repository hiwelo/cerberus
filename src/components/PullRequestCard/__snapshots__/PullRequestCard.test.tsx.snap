// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullRequestCard should match snapshot 1`] = `
.c5 {
  position: relative;
  color: #006e97;
  font-weight: 500;
}

.c5:focus {
  outline: none;
}

.c9 {
  display: inline-block;
  padding: 0.8rem 1.6rem 0.7rem;
  background-color: transparent;
  border: 0.3rem solid #006e97;
  border-radius: 0.8rem;
  color: #222222;
  font-weight: 500;
  line-height: 1.2em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: background-color 0.3s ease-in-out,color 0.2s ease-in-out;
  transition: background-color 0.3s ease-in-out,color 0.2s ease-in-out;
}

.c9:focus,
.c9:hover {
  background-color: #006e97;
  color: #fafaff;
}

.c8 {
  margin-bottom: 1.6rem;
}

.c8:not(:last-of-type) {
  margin-right: 1.6rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0;
  list-style: none;
}

.c6 {
  position: relative;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6::before {
  position: absolute;
  bottom: 0.1em;
  left: 0;
  right: 0;
  content: '';
  height: 0.15em;
  background-color: #e8c000;
  border-radius: 0.4rem;
}

.c6:focus {
  outline: none;
}

.c6:focus::before,
.c6:hover::before {
  background-color: #8ab0ab;
  bottom: 0;
}

.c0 {
  margin-bottom: 0.8rem;
}

.c3 {
  margin-right: 0.4em;
  font-weight: 300;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 1.6rem;
  font-size: 1.4rem;
}

.c4 {
  margin: 0;
  margin-right: 1.6rem;
  font-weight: 400;
}

.c1 {
  display: inline-block;
  margin: 0;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-letter-spacing: -0.005em;
  -moz-letter-spacing: -0.005em;
  -ms-letter-spacing: -0.005em;
  letter-spacing: -0.005em;
}

@media (prefers-reduced-motion:no-preference) {
  .c6::before {
    -webkit-transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
    transition: background-color 0.2s ease-in-out,bottom 0.1s ease-in-out;
  }
}

<article
  class=""
>
  <header
    class="c0"
  >
    <h3
      class="c1"
    >
      Test of pull request
    </h3>
  </header>
  <dl
    class="c2"
  >
    <dt
      class="c3"
    >
      PR
    </dt>
    <dd
      class="c4"
    >
      <a
        class="c5 c6"
        href="https://github.com/owner/repository/pull/123"
        rel="noopener noreferrer"
        target="_blank"
      >
        #123
      </a>
    </dd>
    <dt
      class="c3"
    >
      Author
    </dt>
    <dd
      class="c4"
    >
      <a
        class="c5 c6"
        href="https://github.com/test"
        rel="noopener noreferrer"
        target="_blank"
      >
        test
      </a>
    </dd>
  </dl>
  <div
    class=""
  >
    <ul
      class="c7"
    >
      <li
        class="c8"
      >
        <a
          class="c5 c9"
          href="https://example.com/"
        >
          test
        </a>
      </li>
    </ul>
  </div>
</article>
`;
